internal/services/fingerprint/consumer.go

```diff
@@ -122,7 +147,54 @@ func (c *Consumer) Handle(ctx context.Context, event *Event) error {
 	}
 
 	if observedVIN != vn.Vin {
-		c.logger.Warn().Msgf("Observed VIN %s for vehicle %d with VIN %s.", observedVIN, vn.TokenID, vn.Vin)
+		c.logger.Warn().Str("device", vn.UserDeviceID.String).Str("verified-vin", vn.Vin).Str("observed-vin", observedVIN).Msg("invalid vin")
+		return nil
+	}
+
+	if vc := vn.R.Claim; vc != nil {
+		weekEnd := NumToWeekEnd(GetWeekNum(time.Now()))
+		if vc.ExpirationDate.After(weekEnd) {
+			return nil
+		}
+	}
+
+	if _, err := c.iss.VIN(observedVIN, vn.TokenID.Int(nil), time.Now().Add(8*24*time.Hour)); err != nil {
+		return err
+	}
+
+	c.logger.Info().Str("device-addr", event.Subject).Msg("issued vin credential")
+
+	return nil
+}
+
+func (c *Consumer) HandleSyntheticFingerprint(ctx context.Context, event *Event, key string) error {
+	if !common.IsHexAddress(event.Subject) {
+		return fmt.Errorf("subject %q not a valid address", event.Subject)
+	}
+
+	observedVIN, err := services.ExtractVIN(event.Data)
```

Comment:
I included `week` in the payload because I was worried about messages coming in around the cutover, combined with clock skew. Does that make any sense?